### ISSUE TITLE
Make "comma-style" support {Array,Object}Pattern and function arguments

### ISSUE THREAD

**Issue by maxenglander** [eslint/eslint#7470]
**What version are you using?**

```
 $ eslint --version
 v3.8.1
```

**What did you do?**

```
$ cat | eslint --env es6  \
>              --parser-options '{"ecmaVersion":7,"sourceType":"module"}' \
>              --rule '{"comma-style":["error","last"]}' --no-eslintrc --stdin
const foo = ([
    a
  , b
  , c
]) => {
    return a * b * c;
}

foo(1
  , 2
  , 3);

const bar = ({
    a
  , b
  , c
}) => {
    return `${a} then ${b} then ${c}`;
}^D
```

**What happened?**

No errors were reported by eslint.

**What did you expect to happen?**

Would like it if comma-style violations were reported for the `foo` and `bar` definitions, as well as the `foo` invocation.

**Comment by not-an-aardvark** [comment]
Thanks, I can reproduce this. This seems like a bug to me.

**Comment by maxenglander** [comment]
Thanks @not-an-aardvark I submitted a PR filed as an Update.

**Comment by mysticatea** [comment]
This seems like a bug to me, too.

@eslint/eslint-team Can we change the default behavior in minor-bump? I think this is a bug fix which increases warnings as similar to https://github.com/eslint/eslint/pull/7332.

**Comment by kaicataldo** [comment]
I agree that the expected behavior would be for this to check without a flag and think it should be a semver-minor bug fix.

**Comment by not-an-aardvark** [comment]
I noticed that it wasn't mentioned explicitly in this issue (although it is mentioned in @maxenglander's commit), but we should make sure that comma style can be enforced for imports as well:

```js
import {
  foo
  , bar
  , baz
} from 'qux.js'.
```

Also see: https://github.com/eslint/eslint/issues/7551
